### PR TITLE
Save direct deposit pt2

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

This is a chonky PR, I apologize. Working on getting this across the finish line with lots of unit tests, tweaks, and fixes

- Some of the maintenance changes:
  - Removed the ChildrenDetails and related test for the DevTools widget. It wasn't useful and led to some errors.
  - DevTools now just renders as a 'panel' instead of optionally a modal UI. The modal isn't great and users of the DevTools component have preferred the panel in general.
  - Cleaned up DevToolsLoader and related unit test
 
**IMPORTANT CHANGE TO THE PROFILE APP IN GENERAL**
Decided to wrap the Profile application in StrictMode. This is how React is supposed to be used, and it should show some potential bugs when running locally if they occur. You can read more about it on the React docs: https://react.dev/reference/react/StrictMode 
NOTE: THIS DOES CAUSE EFFECTS AND RENDERING TO HAPPEN A SECOND TIME IN DEV / LOCAL I don't know how many times I tried to track down a double render or double api call effect that turned out to be just the strict mode double running things. I am hoping that the pros outweigh the cons, but if it leads to headaches we can always remove it.
 
- Direct Deposit based changes
  - use VaButton where possible instead of standard button elements
  - handle multiple errors from PUT response
  - move some logic from the AccountUpdateView into the useDirectDeposit hook
  - standardize imports for DirectDeposit root component
  - make sure headline renders consistently for 'Direct deposit information' regardless of what state the page loads into
  - instead of passing individual props to the various components, we can just spread all the results from useDirectDeposit into them, so I did that to help with maintaining any future additions to the hook results
  - changed formIsDirty to hasUnsavedFormEdits for clarity
  - tried to comment on the core useDirectDeposit logic to help with documenting what it all does

- added the useDirectDepositEffects hook to run all the effects that are needed on the page
  - this allows the use of the useDirectDeposit hook anywhere in the application and not have to worry about effects getting duplicated or other bugs around those effects being created

- Added ineligible alert  

- I created a 'DirectDepositDevWidget' that wraps the DevTools, because I could do a few more things with it and now when running locally you have a handy "Auto fill form" that I used when deving out feature 😄 

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/79307

## Testing done

- Lots of tests added
- Feeling pretty good about the hook and component test coverage in general (E2E tests coming next)

Screenshot 2024-04-08 at 11 37 13 PM

Screenshot 2024-04-08 at 11 38 46 PM

## Screenshots

Believe it or not, there aren't a huge amount of UI changes to document.

Fancy new direct deposit dev tool widget that we are going to user for UAT testing (only shows up for specific users with the feature toggle turned on, and 'Auto fill form' button only shows up locally) :
Screenshot 2024-04-08 at 11 39 46 PM

## What areas of the site does it impact?

Profile -> Direct Deposit, behind feature toggle

## Acceptance criteria

- [x] Tests pass
- [x] 🤷🏼 
- [x] Form updates correctly, and using other mock responses shows error alerts in various cases (E2E tests will run through most of these cases, but you can check em out locally by uncommenting all the `direct_deposits` sample responses for GET and PUT requests

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed

### Error Handling

- [x] Browser console contains no warnings or errors.

### CODE REVIEW

File: src/applications/personalization/common/components/devtools/DevToolsLoader.jsx [hunk 1]
```diff
@@ -37,76 +34,35 @@ const DevToolsPanel = ({ devToolsData, show, setShow, panel, children }) => {
...
-          onCloseEvent={handlers.close}
-          onPrimaryButtonClick={function noRefCheck() {}}
-          onSecondaryButtonClick={handlers.close}
-          primaryButtonText="Copy to clipboard"
-          secondaryButtonText="Close"
-          visible
-          clickToClose
+    <div className={classes} data-testid="devtools-panel">
+      <div className="devtools-panel__content">
+        <button
```
Comment (powellkerry) [inline src/applications/personalization/common/components/devtools/DevToolsLoader.jsx:39]: Is there a reason for not using va-button?

File: src/applications/personalization/common/components/devtools/DevToolsLoader.jsx [hunk 2]
```diff
@@ -148,23 +104,22 @@ export const DevToolsLoader = ({
...
     <div className="devtools-container">
-      <button
-        type="button"
-        className="devtools-show-button vads-u-background-color--primary vads-u-color--white"
-        onClick={handlers.togglePanel}
-      >
-        <i className="fas fa-code" />
-      </button>
+      {showIcon && (
+        <button
```
Comment (powellkerry) [inline src/applications/personalization/common/components/devtools/DevToolsLoader.jsx:108]: Is there a reason for not using va-button?

**Review by allisonlu — APPROVED** [review]
🎉 :shipit: ✨

File: src/applications/personalization/common/components/devtools/DevToolsLoader.jsx (hunk 2 above)
Comment (adamwhitlock1) [reply src/applications/personalization/common/components/devtools/DevToolsLoader.jsx:108]: I think I had issues with the first tests with the va-button. I'll revisit that and see if I can move to use the va-button instead. Thanks

**Pull request merged**